src/bmp/message.rs

```diff
@@ -1105,19 +1097,19 @@ impl<'a> Display for InformationTlv<'a> {
 
 }
 
-/// Types of Information TLVs.
-///
-/// See also
-/// <https://www.iana.org/assignments/bmp-parameters/bmp-parameters.xhtml#initiation-peer-up-tlvs>
-#[derive(Debug, Eq, PartialEq)]
-pub enum InformationTlvType {
-    String,         // type 0
-    SysDesc,        // type 1
-    SysName,        // type 2
-    VrfTableName,   // type 3, RFC 9069
-    AdminLabel,     // type 4, RFC 8671
-    Undefined(u16),
-}
+typeenum!(
+    /// Types of Information TLVs.
+    ///
+    /// See also
+    /// <https://www.iana.org/assignments/bmp-parameters/bmp-parameters.xhtml#initiation-peer-up-tlvs>
+    InformationTlvType, u16,
+    0 => String,
+    1 => SysDesc,
+    2 => SysName,
+    3 => VrfTableName,
+    4 => AdminLabel;
+    5.. => Undefined
+);
```

Comment:
Unless we have it elsewhere, keeping undefined values around might be handy. Instead this PR goes the other way and discards a value we previously kept. Not sure about that at first glance.

Reply:
It also loses the RFC references which might be handy to keep.

Reply:
It doesn't discard the value, it passes it in the enum variant (that's not expressed in the macro)

Reply:
Re the RFC reference: You could extend the macro to also allow per-variant attributes.